packages/connect/src/Detector.ts

```diff
@@ -159,31 +134,34 @@ export class Detector {
    * @remarks
    * This is used internally for advanced PolkadotJS use cases and is not supported.  Use {@link connect} instead.
    */
-  public provider = (
+  public provider = async (
     chain: ChainInfo,
     parachainSpec?: string,
-  ): ProviderInterface => {
-    let provider: ProviderInterface = {} as ProviderInterface
-
-    if (!chain.name && !Object.keys(this.#chainSpecs).includes(chain.name)) {
+  ): Promise<ProviderInterface> => {
+    if (!chain.name && !SupportedChains[chain.name as SupportedChains]) {
```

Comment:
The API user **must** be able to use a chain other than the support ones by providing their own JSON.